assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -0,0 +1,558 @@
+							};
+						} ),
+						metricValues: [ ...( rows[ 0 ]?.metricValues || [] ) ],
+					},
+				].concat(
+					hasDateRange
+						? [
+								{
+									dimensionValues: dimensions.map(
+										( dimension ) => {
+											if ( dimension === 'dateRange' ) {
+												return {
+													value: 'date_range_1',
+												};
+											}
+
+											return {
+												value: 'RESERVED_MIN',
+											};
+										}
+									),
+									metricValues: [
+										...( rows[ 0 ]?.metricValues || [] ),
+									],
+								},
+						  ]
+						: []
+				);
+			}
+
+			if ( metricAggregations.includes( 'MAXIMUM' ) ) {
+				data.maximums = [
+					{
+						dimensionValues: dimensions.map( ( dimension ) => {
+							if ( dimension === 'dateRange' ) {
+								return { value: 'date_range_0' };
+							}
+
+							return {
+								value: 'RESERVED_MAX',
+							};
+						} ),
+						metricValues: [
+							...( rows[ rows.length - 1 ]?.metricValues || [] ),
+						],
+					},
+				].concat(
+					hasDateRange
+						? [
+								{
+									dimensionValues: dimensions.map(
+										( dimension ) => {
+											if ( dimension === 'dateRange' ) {
+												return {
+													value: 'date_range_1',
+												};
+											}
+
+											return {
+												value: 'RESERVED_MAX',
+											};
+										}
+									),
+									metricValues: [
+										...( rows[ rows.length - 1 ]
+											?.metricValues || [] ),
+									],
+								},
+						  ]
+						: []
+				);
+			}
+
+			if ( metricAggregations.includes( 'TOTAL' ) ) {
+				// Same here, we pretend that the last row contains totals because we don't need it to be mathematically valid.
+				data.totals = [
+					{
+						dimensionValues: dimensions.map( ( dimension ) => {
+							if ( dimension === 'dateRange' ) {
+								return { value: 'date_range_0' };
+							}
+
+							return {
+								value: 'RESERVED_TOTAL',
+							};
+						} ),
+						metricValues: [
+							...( rows[ rows.length - 1 ]?.metricValues || [] ),
+						],
+					},
+				].concat(
+					hasDateRange
+						? [
+								{
+									dimensionValues: dimensions.map(
+										( dimension ) => {
+											if ( dimension === 'dateRange' ) {
+												return {
+													value: 'date_range_1',
+												};
+											}
+
+											return {
+												value: 'RESERVED_TOTAL',
+											};
+										}
+									),
+									metricValues: [
+										...( rows[ rows.length - 1 ]
+											?.metricValues || [] ),
+									],
+								},
+						  ]
+						: []
+				);
+			}
+		} );
+
+	// Set the original seed value for the faker.
+	faker.seed( originalSeedValue );
+
+	return [
+		{
+			dimensionHeaders:
+				args?.dimensions?.map( ( dimension ) => ( {
+					name: dimension,
+				} ) ) || null,
+			metricHeaders: validMetrics.map( ( metric ) => ( {
+				name: metric?.name || metric.toString(),
+				type: `TYPE_${ getMetricType( metric ) }`,
```

Comment:
As mentioned above, it would be preferable to avoid this string concatenation and instead use the full string values throughout.

```suggestion
				type: `TYPE_${ getMetricType( metric ) }`,
```

Reply:
@techanvil Here we are dynamically getting the type, so we still need to call the `getMetricType` function but without the `TYPE_` prefix.

Reply:
That sounds fine to me, my point was really just a reminder to update following the comment above.